--- origsrc/gst-plugins-espeak-0.3.3/configure.ac	2009-04-29 14:48:43.000000000 -0500
+++ src/gst-plugins-espeak-0.3.3/configure.ac	2010-03-06 22:37:44.533968200 -0600
@@ -123,8 +123,13 @@ else
 fi
 AC_SUBST(plugindir)
 
+dnl LDFLAGS really should only contain flags, not libs - they get added before
+dnl whatevertarget_LIBS and -L flags here affect the rest of the linking
+GST_ALL_LDFLAGS="-no-undefined"
+AC_SUBST(GST_ALL_LDFLAGS)
+
 dnl set proper LDFLAGS for plugins
-GST_PLUGIN_LDFLAGS='-module -avoid-version -export-symbols-regex [_]*\(gst_\|Gst\|GST_\).*'
+GST_PLUGIN_LDFLAGS="-module -avoid-version -export-symbols-regex '^[_]*\(gst_plugin_desc\$\$' $GST_ALL_LDFLAGS"
 AC_SUBST(GST_PLUGIN_LDFLAGS)
 
 dnl check espeak dependencies
